docs: improve readme and add REACT_APP_ORDER_BOOK_URLS

anxolin opened the pull request:
# Summary

This PR just moves arrounds some elements in the documentation to make it a bit more structured.

Additionally, documents `REACT_APP_ORDER_BOOK_URLS` env var, to make it easier to use your own API
https://cowservices.slack.com/archives/C036G0J90BU/p1695996920984219

fleupold commented on README.md:
```diff
@@ -96,30 +98,41 @@ yarn ui:build
+
+The app has some default configuration (in `.env` and `.env.prod` file), but it's highly encouraged to define your own `.env.local`.
 
 Make a copy of `.env` named `.env.local`, this will allow you to set your own configuration only in your local environment.
```
```suggestion
Inside `apps/cowswap-frontend`, make a copy of `.env` named `.env.local`, this will allow you to set your own configuration only in your local environment.
```

fleupold reviewed (approved):
Thanks

shoom3301 reviewed (approved):
Thank you!

alfetopito commented on README.md:
````diff
@@ -96,30 +98,41 @@ yarn ui:build
-`INTEGRATION_TESTS_PRIVATE_KEY`=YOUR_TEST_WALLET_PRIVATE_KEY
+```ini
+# Define your own RPC endpoints
+REACT_APP_NETWORK_URL_1=https://mainnet.infura.io/v3/{YOUR_INFURA_KEY}
+REACT_APP_NETWORK_URL_5=https://goerli.infura.io/v3/{YOUR_INFURA_KEY}
+REACT_APP_NETWORK_URL_100=https://rpc.gnosischain.com
````
Since you mention the INFURA key down below, I think these examples could be generic. What do you think?

```suggestion
REACT_APP_NETWORK_URL_1=https://...  
REACT_APP_NETWORK_URL_5=https://...
REACT_APP_NETWORK_URL_100=https://...
```

anxolin replied:
I will rebase and merge into this other PR changes, so i do all at once

anxolin replied:
Fixed

alfetopito commented on README.md:
````diff
@@ -142,52 +155,38 @@ The format of the JSON follows this typescript format: <src/utils/metadata.ts>
-REACT_APP_NETWORK_URL_5=https://goerli.infura.io/v3/{YOUR_INFURA_KEY}
-REACT_APP_NETWORK_URL_100=https://rpc.gnosischain.com
-```
+The reference implementation of the API is [gp-v2-services](https://github.com/cowprotocol/services).
````
```suggestion
The reference implementation of the API is [CoW Protocol Services](https://github.com/cowprotocol/services).
```

alfetopito commented on README.md:
```diff
@@ -142,52 +155,38 @@ The format of the JSON follows this typescript format: <src/utils/metadata.ts>
-
-Fee quote requests and posting orders are sent to an API. This API has the responsibility of collecting orders and
-handing them to the solvers.
+The app uses a Wallet Connect v1 bridge.
```
Not anymore. We have removed it, and this section should be removed as well

anxolin replied:
fixed

anxolin commented:
Addressed comments! I also rebased with https://github.com/cowprotocol/cowswap/pull/3276 which cleans the README a bit, so I will deliver the 2 PRs together.
